odl/discr/discretization.py

```diff
@@ -316,6 +316,26 @@ def __setitem__(self, indices, values):
         else:
             self.ntuple.__setitem__(indices, values)
 
+    @property
+    def restriction(self):
+        """The restriction operator associated with this vector.
+
+        See Also
+        --------
+        RawDiscretization.restriction : For full description
+        """
+        return self.space.restriction(self.ntuple)
+
+    @property
+    def extension(self):
+        """The extension operator associated with this vector.
+
+        See Also
+        --------
+        RawDiscretization.extension : For full description
+        """
+        return self.space.extension(self.ntuple)
```

Comment:
> and that one should be quite fast, no?

For the current extensions we have, this is instantaneous since it just creates the interpolator and returns it. I was more thinking of possible future applications, where this extension can involve significant computations (e.g. calculating a tessellation?). The syntax is nice, admittedly, but we may have to change it later and should consider the slightly more ugly non-property syntax.